Source/carthage/CopyFrameworks.swift

```diff
@@ -18,29 +18,61 @@ public struct CopyFrameworksCommand: CommandType {
-		}
+		return producerWithOptions(CopyFrameworksOptions.evaluate(mode))
+			|> flatMap(.Merge) { options -> SignalProducer<(), CommandError> in
+				return inputFiles()
+					|> flatMap(.Concat) { frameworkPath -> SignalProducer<(), CarthageError> in
+						let frameworkName = frameworkPath.lastPathComponent
+
+						let source = Result(NSURL(fileURLWithPath: frameworkPath, isDirectory: true), failWith: CarthageError.InvalidArgument(description: "Could not find framework \"\(frameworkName)\" at path \(frameworkPath). Ensure that the given path is appropriately entered and that your \"Input Files\" have been entered correctly."))
+						let target = frameworksFolder().map { $0.URLByAppendingPathComponent(frameworkName, isDirectory: true) }
+
+						return combineLatest(SignalProducer(result: source), SignalProducer(result: target), SignalProducer(result: validArchitectures()))
+							|> flatMap(.Merge) { (source, target, validArchitectures) -> SignalProducer<(), CarthageError> in
+								return combineLatest(copyProduct(source, target), codeSigningIdentity())
+									|> flatMap(.Merge) { (url, codesigningIdentity) -> SignalProducer<(), CarthageError> in
+										let strip = stripFramework(url, keepingArchitectures: validArchitectures, codesigningIdentity: codesigningIdentity)
+										if options.copyBCSymbolMaps {
+											return strip
+												|> then(self.copyBCSymbolMapsForFramework(url, fromDirectory: source.URLByDeletingLastPathComponent!))
+												|> then(.empty)
+										} else {
+											return strip
+										}
+									}
+							}
+					}
+					|> promoteErrors
+			}
+			|> waitOnCommand
+	}
+	
+	private func copyBCSymbolMapsForFramework(frameworkURL: NSURL, fromDirectory directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+		return UUIDsForFramework(frameworkURL)
+			|> flatMap(.Merge) { uuids in SignalProducer(values: uuids) }
+			|> map { uuid in directoryURL.URLByAppendingPathComponent(uuid.UUIDString).URLByAppendingPathExtension("bcsymbolmap") }
+			|> filter { url in url.checkResourceIsReachableAndReturnError(nil) }
+			|> flatMap(.Merge) { url in
+				return SignalProducer(result: builtProductsFolder())
+					|> flatMap(.Merge) { builtProducts in
+						let target = builtProducts.URLByAppendingPathComponent(url.lastPathComponent!)
+						return copyProduct(url, target)
+					}
+			}
+	}
+}
+
+private struct CopyFrameworksOptions: OptionsType {
+	let copyBCSymbolMaps: Bool
+	
+	static func create(copyBCSymbolMaps: Bool) -> CopyFrameworksOptions {
+		return self(copyBCSymbolMaps: copyBCSymbolMaps)
+	}
+	
+	static func evaluate(m: CommandMode) -> Result<CopyFrameworksOptions, CommandantError<CarthageError>> {
+		let actionIsArchive = buildAction() == "install" // archives use ACTION=install
+		return create
+			<*> m <| Option(key: "copy-bcsymbolmaps", defaultValue: actionIsArchive, usage: "copy the bcsymbolmap files for each product built with bitcode (enabled by default for archive builds)")
```

Comment:
I came to the thought that would be unnecessary.

> maybe they have a shell script phase after copy-frameworks that strips the bitcode out of all the frameworks

I don't know whether there is a way to do that (stripping bitcode) or not.

> After considering it some more, I think it's plausible that someone might be building an IPA file without using Xcode's archive command, which means they'd want a way to opt-in to copying the bcsymbolmaps.

Taking it into consideration, we should always copy bcsymbolmaps regardless of the build action. Maybe that would not have any bad side effects?

> The other reason I wanted the option was because it's a convenient place to document that carthage copy-frameworks will copy bcsymbolmaps on archive builds. Putting that in the command description makes an already-extremely-long description even more ridiculously long.

This seems not to be a valid reason for adding the option / complexity.